### ISSUE TITLE
BUG: TreeEnsemble.predict() checks "logloss" instead of "log_loss", y-validation never triggers

### ISSUE THREAD

**Issue by ojuschugh1** [shap/shap#4828]
### Issue Description

`TreeEnsemble.predict()` on line 1708 checks `output == "logloss"` but the actual value passed in is `"log_loss"` (with underscore). Every other place in the file uses `"log_loss"`. Because of this mismatch, the y-validation block is dead code and never runs.

This means calling `predict(X, output="log_loss")` without passing `y` silently returns results instead of telling you that labels are required.

This was originally reported in #1205 back in May 2020 but got closed by the stale bot without being fixed. The bug is still present on master.

The fix is one line: change `"logloss"` to `"log_loss"` on line 1708.

### Minimal Reproducible Example

```python
import numpy as np
from sklearn.ensemble import GradientBoostingRegressor
import shap

model = GradientBoostingRegressor(n_estimators=10, random_state=0)
X = np.random.randn(100, 4)
y = np.random.randn(100)
model.fit(X, y)

explainer = shap.TreeExplainer(model, X[:50], model_output="log_loss")

# This should raise ValueError because log_loss needs labels (y)
# But it silently returns results because line 1708 checks "logloss" not "log_loss"
result = explainer.model.predict(X[:5], output="log_loss")
print(f"predict returned shape {result.shape} without y — should have raised an error")
```

### Traceback

```shell
No traceback - that's the bug. The call succeeds silently when it should raise:


predict returned shape (5,) without y — should have raised an error
```

### Expected Behavior

`predict(X, output="log_loss")` without `y` should raise `ValueError` saying labels are required, same as `_validate_inputs` does on line 471.

The fix:

```python
# line 1708, currently:
if output == "logloss":

# should be:
if output == "log_loss":
```

### Bug report checklist

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [x] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [x] I'd be interested in making a PR to fix this bug

### Installed Versions

```
shap: 0.51.1.dev49 (master branch)
python: 3.12.12
numpy: 2.3.5
```
